dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java

```diff
@@ -1245,6 +1274,69 @@ public void Test_Pagination_And_Sort_Then_Request_Filter_Expect_Empty_Results()
+     */
+     @Test
+     public void Test_Export_Import() throws IOException {
+         final AppDescriptorDataGen dataGen = new AppDescriptorDataGen()
+                 .stringParam("param1", false,  true)
+                 .stringParam("param2", false,  true)
+                 .withName("import-export-test")
+                 .withDescription("import-export-test")
+                 //We're indicating that extra params are allowed to test required params are still required.
+                 .withExtraParameters(true);
+         final String key = dataGen.getKey();
+         final HttpServletRequest request = mock(HttpServletRequest.class);
+         final HttpServletResponse response = mock(HttpServletResponse.class);
+
+         when(request.getRequestURI()).thenReturn("/baseURL");
+         final String fileName = dataGen.getFileName();
+         final File file = dataGen.nextPersistedDescriptor();
+         try(InputStream inputStream = Files.newInputStream(file.toPath())){
+             final FormDataMultiPart formDataMultiPart = createFormDataMultiPart(fileName, inputStream);
+             final Response appResponseOk = appsResource.createApp(request, response,formDataMultiPart);
+             Assert.assertNotNull(appResponseOk);
+             Assert.assertEquals(HttpStatus.SC_OK, appResponseOk.getStatus());
+
+             final Response availableAppsResponse = appsResource
+                     .listAvailableApps(request, response, null);
+             Assert.assertEquals(HttpStatus.SC_OK, availableAppsResponse.getStatus());
+             final ResponseEntityView responseEntityView1 = (ResponseEntityView) availableAppsResponse
+                     .getEntity();
+             final List<AppView> integrationViewList = (List<AppView>) responseEntityView1
+                     .getEntity();
+             assertFalse(integrationViewList.isEmpty());
+
+             final Host site = new SiteDataGen().nextPersisted();
+
+             final String siteId = site.getIdentifier();
+             final Map<String, Input> inputParamMap = ImmutableMap.of(
+              "param1", newInputParam("value".toCharArray()),
+              "param2", newInputParam("value".toCharArray())
+             );
+             final SecretForm secretForm1 = new SecretForm(inputParamMap);
+             final Response createSecretResponse1 = appsResource.createAppSecrets(request, response, key, siteId, secretForm1);
+             Assert.assertEquals(HttpStatus.SC_OK, createSecretResponse1.getStatus());
+
+             final String password = "123456789";
+             ExportSecretForm form = new ExportSecretForm(password,false, ImmutableMap.of(siteId, ImmutableSet.of(key)));
+             final OutboundJaxrsResponse exportResponse = (OutboundJaxrsResponse)appsResource.exportSecrets(request, response, form);
+             final String json = String.format("{ password: \"%s\" }",password);
+             final InputStream entity = (InputStream) exportResponse.getEntity();
```

Comment:
Codacy found an issue: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5931733)